core/trashcan.js

```diff
@@ -96,12 +109,46 @@ Blockly.Trashcan.prototype.SPRITE_LEFT_ = 0;
  */
 Blockly.Trashcan.prototype.SPRITE_TOP_ = 32;
 
+/**
+ * The openness of the lid when the trashcan contains blocks.
+ *    (0.0 = closed, 1.0 = open)
+ * @type {number}
+ * @private
+ */
+Blockly.Trashcan.prototype.HAS_BLOCKS_LID_ANGLE = 0.1;
+
+/**
+ * The maximum number of blocks that can go in the trashcan's flyout. '0' turns
+ * turns off trashcan contents, 'Infinity' sets it to unlimited.
+ * @type {number}
+ */
+Blockly.Trashcan.prototype.MAX_CONTENTS = 32;
```

Comment:
File an issue to eventually set the trashcan max contents through the options struct, and you can do it as a followup PR.